### PR TITLE
Extend email plugin to allow custom actions

### PR DESCRIPTION
**Pull request by ScottAgirs**
Hello Strapi, with this PR I would like to propose exposing an additional resolver to the email-plugin.

It appears that currently there's an issue as there's limited custom functionality possible.

Exposing a method that would allow plugin developers to add custom functionality such as managing email lists, contacts and potentially other concerns related to email providers, IMHO, would bring a big value to this plugin.

I have with this PR added a `strapi.plugins.email.provider.action(options)` resolver as well as brief documentation on how it could work.

If approved, you can expect an extend documentation with examples momentarily. Currently working with Mailjet provider, however, if approved happy to others as well.

🙌

### CODE REVIEW

**Comment by alexandrebodin** [comment]
Hi, I don't really see the point of doing this.
The email provider is really aimed at being compatible with the email plugin and not made to provide extra emailing fatures. I really believe this should be done outside the email plugin provider instead.

I see it like this: Create a custom service with my (let's say mailjet for the example) mailjet client. Enable to setup the email provider with a custom client (would be a nice addon to the provider interface) then add the custom functions I want inside my custom emailing service. 

What do you think ?

**Comment by ScottAgirs** [comment]
> Hi, I don't really see the point of doing this.
> The email provider is really aimed at being compatible with the email plugin and not made to provide extra emailing fatures. I really believe this should be done outside the email plugin provider instead.
> 
> I see it like this: Create a custom service with my (let's say mailjet for the example) mailjet client. Enable to setup the email provider with a custom client (would be a nice addon to the provider interface) then add the custom functions I want inside my custom emailing service.
> 
> What do you think ?

Interesting. 

That would mean installing additional packages, no? 
IMHO, this would add extra complexity which is directly related to the email handling. 

I think I understand what you mean @alexandrebodin by saying that the provider is handling email sending itself, but should we really disable users from any other functionality that the email provider may have? 
It feels a little odd that users would have to stack up multiple layers of packages/plugins in order to take care of essentially the same service (please let me know if this makes sense or if I'm missing something).

In short: I think it would be really nice to have a streamlined way to programmatically handle the extra functionality of a given email provider to expose methods that are close to their API.

Any ideas on how could we accomplish this without installing additional packages with this setup?

**Comment by alexandrebodin** [comment]
Hi,

I would split the logic into two distinc things: a provider and a service with extra features. We would need to allow the registration of both of them via a service container / plugin system of some sort.

Right now if your provider provides extra methods you can still document it in your provider and tell user to call them using strapi.plugins.email.provider[theExtraMethod]() without adding an extra API to Strapi.

**Comment by alexandrebodin** [comment]
Hi @ScottAgirs as we didn't have more discussion and you can already achieve what you want without having to run specific logic inside of Strapi I'll close this PR. Let us know if you feel there is still something to do inside Strapi :)
